### PR TITLE
Generics improvements

### PR DESCRIPTION
**Pull request by kbuntrock**
- Handles objects extending Map or Collection.
- Handles Object.class description

### CODE REVIEW

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/configuration/library/reader/JakartaRsReader.java [hunk 1]
```diff
@@ -157,7 +153,7 @@ protected List<ParameterObject> readParameters(final Method originalMethod, fina
 				logger.debug("Parameter : " + parameter.getName());
 
 				ParameterObject paramObj = new ParameterObject(parameter.getName(),
-					genericityResolver.getContextualType(parameter.getParameterizedType(), method));
+					genericityResolver.resolve(clazz, parameter.getParameterizedType()));
```
Comment (murdos) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/configuration/library/reader/JakartaRsReader.java:156]: How about using `method.getDeclaringClass()`? Would that works?

File: openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/generic_parent_bound_by_child.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (murdos) [inline openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/generic_parent_bound_by_child.yml]: I think that this file, and the other one `parametrized_object_class.yml,`are not used

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/configuration/library/reader/JakartaRsReader.java (hunk 1 above)
Comment (kbuntrock) [reply openapi-maven-plugin/src/main/java/io/github/kbuntrock/configuration/library/reader/JakartaRsReader.java:156]: No, that was the previous way to resolve it and it was not working that well.

Previously, in presence of a inheritance chain, I firstly creating a reference "array" object. Starting from the child and going up each link ; saving for this link the parametrized name and the actual type corresponding to that name.

Then, when I was resolving a method parameter or response, I was reading this linked representation starting from the method declaring class, and going down till the actual type was resolved.

This was not exhaustive, I was putting aside a lot of cases in creating this reference object.

Turns out Guava is much better in this than me. I totally deleted my first horrible attempt to handle this manually, and I let the "TypeToken" resolver do its work.

File: openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/generic_parent_bound_by_child.yml (hunk 2 above)
Comment (kbuntrock) [reply openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/generic_parent_bound_by_child.yml]: Right, thank you. :)

**Pull request merged**